kernel/kernel-api/src/main/java/io/delta/kernel/internal/clustering/ClusteringMetadataDomain.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.clustering;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.delta.kernel.expressions.Column;
+import io.delta.kernel.internal.SnapshotImpl;
+import io.delta.kernel.internal.metadatadomain.JsonMetadataDomain;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/** Represents the metadata domain for clustering. */
+public final class ClusteringMetadataDomain extends JsonMetadataDomain {
+
+  public static final String DOMAIN_NAME = "delta.clustering";
+
+  /**
+   * Constructs a ClusteringMetadataDomain with the specified physical clustering columns and
```

Comment:
Just to be clear, by `physical` clustering columns, do you mean that these is the physical column name?

i.e. this relates to ColumnMapping?

Reply:
Define specifically as the column names used when writing the data to data files. If column mapping is enabled, use the physical name assigned; otherwise, use the logical column name.